### DISCUSSION TITLE
Rounded size?

### DISCUSSION THREAD

**Discussion by zaferyumruk** [kernc/backtesting.py#602] [Q&A]
Why the following constraint exists in a design perspective? What is the benefit of using rounded size? 
-1 < size < 1 or round(size) == size

**Comment by kernc** [comment]
Do you mean [`Order.size`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Order.size)?
https://github.com/kernc/backtesting.py/blob/8b83b3541a5f00f3d86047caf1997daa966e6dee/backtesting/backtesting.py#L208-L209
If fractional units were allowed (e.g. 3.6 units) only for >1, this would create a weird dichotomy with interval (0, 1), which is interpreted as "proportional share of current equity" (e.g. 5%). It was an up-front design decision.

See also: https://github.com/kernc/backtesting.py/issues/134

**Reply by zaferyumruk** [reply]
Thanks for the clarification. I was following exactly the same workaround you suggested in #134 .
